test/dymaptic.GeoBlazor.Core.Test.ConstructorTesting/Program.cs

```diff
@@ -0,0 +1,254 @@
+            .ToList();
+
+        // Only check classes with 2 or more constructors
+        if (constructors.Count < 2) return null;
+
+        // Check for compliant parameterless constructor
+        var hasCompliantParameterlessConstructor = constructors.Any(IsCompliantParameterlessConstructor);
+
+        if (!hasCompliantParameterlessConstructor)
+        {
+            var relativePath = Path.GetRelativePath(Path.GetDirectoryName(solutionPath)!, filePath);
+            return (classSymbol.Name, relativePath);
+        }
+
+        return null;
+    }
+
+    /// <summary>
+    /// Checks if a constructor is parameterless and has the required attribute.
+    /// </summary>
+    static bool IsCompliantParameterlessConstructor(IMethodSymbol constructor)
+    {
+        return constructor.Parameters.Length == 0 &&
+               HasActivatorUtilitiesConstructorAttribute(constructor);
+    }
+
+    /// <summary>
+    /// Reports the analysis results to the console.
+    /// </summary>
+    static int ReportResults(List<(string className, string filePath)> nonCompliantClasses)
+    {
+        if (nonCompliantClasses.Count == 0)
+        {
+            Console.WriteLine(SUCCESS_MESSAGE);
+            return 0;
+        }
+
+        Console.WriteLine(FAILURE_MESSAGE);
+        foreach (var (className, filePath) in nonCompliantClasses)
+        {
+            Console.WriteLine($"- {className} - {filePath}");
+        }
+        return 1;
+    }
+
+    /// <summary>
+    /// Finds the solution file by traversing up from the current directory.
+    /// </summary>
+    static string FindSolutionFile(string startDirectory)
+    {
+        var directory = new DirectoryInfo(startDirectory);
+
+        while (directory != null)
+        {
+            // Look for the specific solution file
+            var solutionFile = Path.Combine(directory.FullName, "src", "dymaptic.GeoBlazor.Core.sln");
+            if (File.Exists(solutionFile))
+            {
+                return solutionFile;
+            }
+
+            // Move up one directory
+            directory = directory.Parent;
+        }
+
+        return string.Empty;
+    }
+
+    /// <summary>
+    /// Checks if a class inherits from ComponentBase (directly or through MapComponent).
+    /// Most GeoBlazor components inherit from MapComponent, which inherits from ComponentBase.
+    /// </summary>
+    static bool InheritsFromComponentBase(INamedTypeSymbol classSymbol)
+    {
+        var baseType = classSymbol.BaseType;
+
+        while (baseType != null)
+        {
+            var typeName = baseType.Name;
+
+            // Check for MapComponent first (most common in this codebase)
+            if (typeName == "MapComponent")
+            {
+                return true;
+            }
+
+            // Check for ComponentBase (base class of MapComponent and other Blazor components)
+            if (typeName == "ComponentBase")
+            {
+                return true;
```

Comment:
I think I would like this to throw an exception. I can't think of a scenario where we want non-MapComponent Razor Components in GeoBlazor.